lib/internal/Magento/Framework/Reflection/TypeProcessor.php

```diff
@@ -521,14 +521,149 @@ public function getParamType(ParameterReflection $param)
                 $param->getDeclaringFunction()->name
             ));
         }
-        if ($type == 'array') {
+        if ($type === 'array') {
             // try to determine class, if it's array of objects
             $paramDocBlock = $this->getParamDocBlockTag($param);
             $paramTypes = $paramDocBlock->getTypes();
             $paramType = array_shift($paramTypes);
+
+            $paramType = $this->resolveFullyQualifiedClassName($param->getDeclaringClass(), $paramType);
+
             return strpos($paramType, '[]') !== false ? $paramType : "{$paramType}[]";
         }
-        return $type;
+
+        return $this->resolveFullyQualifiedClassName($param->getDeclaringClass(), $type);
+    }
+
+    /**
+     * Get alias mapping for source class
+     *
+     * @param ClassReflection $sourceClass
+     * @return array
+     */
+    public function getAliasMapping(ClassReflection $sourceClass): array
```

Comment:
Please, consider making methods private, looks like they used only in the private implementation

Reply:
Hi @sidolov ,
it is used in `\Magento\Framework\Reflection\Test\Unit\TypeProcessorTest`. We have two options: remove tests or eject this method to a dedicated service.